File: java/spike-jsr330/src/main/java/com/github/gimmi/spikejsr330/application/AppGuiceServletContextListener.java
package com.github.gimmi.spikejsr330.application;

import java.util.logging.Logger;

import org.slf4j.bridge.SLF4JBridgeHandler;

import com.github.gimmi.spikejsr330.framework.FrameworkModule;
import com.google.inject.Guice;
import com.google.inject.Injector;
import com.google.inject.servlet.GuiceServletContextListener;

public class AppGuiceServletContextListener extends GuiceServletContextListener {
	@Override
	protected Injector getInjector() {
		Logger.getLogger("com.google.inject").addHandler(new SLF4JBridgeHandler());
		return Guice.createInjector(new ApplicationModule(), new FrameworkModule());
	}
}

File: java/spike-scala-servlet/src/main/scala/com/github/gimmi/spikescalaservlet/configuration/AppGuiceServletContextListener.scala
package com.github.gimmi.spikescalaservlet.configuration

import com.google.inject.servlet.GuiceServletContextListener
import com.google.inject.Guice

class AppGuiceServletContextListener extends GuiceServletContextListener {
	def getInjector = Guice.createInjector(new AppServletModule)
}

